x-pack/filebeat/input/httpjson/internal/v2/pagination.go

```diff
@@ -159,6 +160,13 @@ func (iter *pageIterator) getPage() (*response, error) {
 	r.url = *iter.resp.Request.URL
 	r.page = iter.n
 
+	if len(bodyBytes) > 0 && r.header.Get("Content-Type") == "application/x-ndjson" {
+		bodyBytes = append([]byte("{\"ndjson-results\":["), bodyBytes...)
+		bodyBytes = bytes.ReplaceAll(bodyBytes, []byte("\r\n"), []byte(","))
+		bodyBytes = bytes.ReplaceAll(bodyBytes, []byte("\n"), []byte(","))
+		bodyBytes = bytes.TrimRight(bodyBytes, ",")
+		bodyBytes = append(bodyBytes, []byte("]}")...)
+	}
```

Comment:
```suggestion
	if len(bodyBytes) > 0 && r.header.Get("Content-Type") == "application/x-ndjson" {
		bodyBytes = append([]byte("["), bodyBytes...)
		bodyBytes = bytes.ReplaceAll(bodyBytes, []byte("\r\n"), []byte(","))
		bodyBytes = bytes.ReplaceAll(bodyBytes, []byte("\n"), []byte(","))
		bodyBytes = bytes.TrimRight(bodyBytes, ",")
		bodyBytes = append(bodyBytes, ']')
	}
```

Root arrays are supported and automatically split, so just converting the object stream to an array should make it work normally without needing to know about the special case if ndjson is received. WDYT?